src/lxc/storage/lvm.c

```diff
@@ -177,10 +185,18 @@ static int do_lvm_create(const char *path, uint64_t size, const char *thinpool)
 	cmd_args.vg = vg;
 	cmd_args.lv = lv;
 	cmd_args.size = sz;
+	cmd_args.sigwipe = true;
 	TRACE("Creating new lvm storage volume \"%s\" on volume group \"%s\" "
 	      "of size \"%s\"", lv, vg, sz);
 	ret = run_command(cmd_output, sizeof(cmd_output),
 			  lvm_create_exec_wrapper, (void *)&cmd_args);
+
+	//If lvcreate is old and doesn't support signature wiping, try again without it.
+	if (strstr(cmd_output, "Error during parsing of command line") != NULL) {
```

Comment:
Ugh, is that really the only way? That seems like a terrible hack.

Reply:
@brauner I agree its far from ideal. 

My original intention with this was to detect the exit code of 3 (EINVALID_CMD_LINE) which hasn't changed for 15 years:

https://github.com/lvmteam/lvm2/blame/6462e8dffc7ed53973ce796cec18fecc0fb89d77/tools/lvmcmdline.c#L2803
https://github.com/lvmteam/lvm2/blame/0ecf232194986b12c3ae95b3f93e7fdd56158dc1/tools/errors.h#L21

However when I came to implement it, I realised that the `run_command` function's return value was not the exit code, but instead a parsed value where -1 was failure and 0 was OK, because it uses `wait_for_pid`.

https://github.com/lxc/lxc/blob/45265bfdfa170a5aab8c655f665b55c425d2abd6/src/lxc/utils.c#L1654
https://github.com/lxc/lxc/blob/45265bfdfa170a5aab8c655f665b55c425d2abd6/src/lxc/utils.c#L308-L309

What I needed was for an equivalent function, but using the `wait_for_pid_status` function which seems to return the exit code: https://github.com/lxc/lxc/blob/45265bfdfa170a5aab8c655f665b55c425d2abd6/src/lxc/utils.c#L314-L331

Perhaps called `run_command_status`?

In the meantime, I researched whether that return response text had changed much over time, and whether it overlapped with introduction of the wipe signatures feature.

The error response output appears to have changed once in 15 years:

15 years ago it was what it is today: "Error during parsing of command line."
https://github.com/lvmteam/lvm2/blame/0dabe7237c531459792de953879a5c4c6de41c65/tools/lvmcmdline.c#L2677

2 years ago it changed briefly for less than a day to:

https://github.com/lvmteam/lvm2/commit/6471bb2c41cf0e692a8fe7ea00033327359d42fc

Before changing back to:

https://github.com/lvmteam/lvm2/commit/2eaca7ab63345594ccfd7133ddf92729b7e7691c#diff-3e7d4e73e6b87bd5050608a3a4bf2285R2677

The `--wipesignatures` and `-Wy` flags were added five years ago (https://github.com/lvmteam/lvm2/commit/169b4c15864330e3fe47efee7b0def06581b8f08).

So that change overlaps the brief fluctuation in error output.

Additionally, comparing to that string `Error during parsing of command line` rather than the more specific output above it `invalid option -- W` also catches versions that don't support the `--yes` flag which outputs: 

```lvcreate: unrecognized option `--yes`
    Error during parsing of command line.
```

I agree that comparing strings in command output is not a great approach, however I wanted to balance that with a more invasive modification to LXC. I also was trying to avoid having to run lvcreate twice on modern systems (in order to introspect the version).

If you're happy with me adding a new `run_command_status` or doing version introspection on lvcreate then I'm happy to change tack on this. Or maybe I've missed a way to get the exit code from `run_command` in an easier way.

Thanks
Tom